### PR TITLE
Added learning Pathways in Web3 resources and  list of the top 25 Web3 coding challenges 

### LINKED ISSUE #215: [FEATURE] - Introduce Learning Pathways for Web3 Education
### Idea Contribution

- [X] I have read all the feature request issues.
- [X] I'm interested in working on this issue
- [X] I'm part of GSSOC organization

### Explain feature request

The current repository lacks a structured approach for users to navigate through the abundance of resources available for learning Web3 technologies. Introducing learning tracks would enable users, especially beginners, to follow a structured pathway, making it easier for them to grasp foundational concepts and progress to more advanced topics.

### Explain your solution

The proposed solution involves implementing learning tracks categorized based on the user's proficiency level—beginner, intermediate, and advanced. Each learning track would comprise a curated list of resources such as websites, tutorials, courses, and tools. For instance, beginners could start with basic tutorials and foundational concepts, while intermediate learners could explore more in-depth tutorials and engage with the community. Advanced users would have access to resources focusing on advanced topics.
Integrating this feature would enhance the user experience of the repository, making it more accessible and valuable to individuals seeking to learn about Web3 technologies.

### Any alternative approaches/features

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by SkSadaf**
PR Title: Added learning Pathways in Web3 resources and  list of the top 25 Web3 coding challenges 

Fixes Issue: Closes #215, #216

Description: Implemented learning tracks categorized based on the user's proficiency level—beginner, intermediate, and advanced. Each learning track comprises a list of resources such as websites, tutorials, courses, and tools. Added a list of the top 25 Web3 coding challenges. Each challenge is accompanied by a brief description and a link to the respective resource or tutorial. 

Checklist:
- [YES] I have carefully reviewed and adhered to the [contributing guidelines](https://github.com/jfmartinz/ResourceHub/blob/main/CONTRIBUTING.md) before creating this pull request.
- [YES] I followed the prescribed PR title template.

### CODE REVIEW

**Comment by jfmartinz** [comment]
---

**Feedback:**

* **What is Pathway?** If by "Pathway" you mean a step-by-step roadmap for learning Web3 from beginner to advanced levels, please find resources that offer such roadmaps. Include the URL and description in a new "Roadmap" section and create a separate table for it.
* **Duplicate and Included Resources:** Remove the following resources as they are already included in the table:
  * [CryptoZombies](https://cryptozombies.io/)
  * Duplicate resource: [Crypto Zombies 2](https://cryptozombies.io/)

**Resources to be included in the [Website to Learn Web3](https://github.com/jfmartinz/ResourceHub/tree/main/Web3#website-to-learn-web3):**

1. [BuidlGuidl](https://buidlguidl.com/)
2. [ChainShot](https://www.chainshot.com/)
3. [Solidity by Example](https://solidity-by-example.org/)
4. [dApps on Ethereum](https://docs.soliditylang.org/en/v0.8.11/introduction-to-smart-contracts.html)
5. [zkSync Integration](https://zksync.io/)
6. [Optimism](https://optimism.io/)


**New Sections:**

* **Sandbox/IDE for Web3:**
  * [ETH.Build](https://eth.build/)
  * [Remix IDE](https://remix-ide.readthedocs.io/en/latest/index.html)

* **Challenges Resources:**
  * [Ethernaut](https://ethernaut.openzeppelin.com/)

* **Hackathon Resources:**
  * [Chainlink Hackathon](https://chain.link/hackathon)
  * [Gitcoin Bounties](https://gitcoin.co/explorer)

**Note:** The sections "Sandbox/IDE for Web3," "Challenges," and "Hackathon Resources" are not yet included in the project. You can either remove these resources or add new sections under the Web3 category and place the resources in their respective sections. Hackathon resources should be listed in the hackathon section table.

**Format for Adding a New Section and Resource:**
```html
### Hackathon
<table width="100%">
  <tr>
    <th>Resource Name</th>
    <th>Description</th>
  </tr>
  <tr>
    <td><a href="URL HERE">RESOURCE NAME</a></td>
    <td>RESOURCE DESCRIPTION</td>
  </tr>
</table>

### Challenges
<table width="100%">
  <tr>
    <th>Resource Name</th>
    <th>Description</th>
  </tr>
  <tr>
    <td><a href="URL HERE">RESOURCE NAME</a></td>
    <td>RESOURCE DESCRIPTION</td>
  </tr>
</table>

### IDE
<table width="100%">
  <tr>
    <th>Resource Name</th>
    <th>Description</th>
  </tr>
  <tr>
    <td><a href="URL HERE">RESOURCE NAME</a></td>
    <td>RESOURCE DESCRIPTION</td>
  </tr>
</table>

### Roadmap
<table width="100%">
  <tr>
    <th>Resource Name</th>
    <th>Description</th>
  </tr>
  <tr>
    <td><a href="URL HERE">RESOURCE NAME</a></td>
    <td>RESOURCE DESCRIPTION</td>
  </tr>
</table>
```
---

if you have any question, dont hesitate.

**Comment by SkSadaf** [comment]
Sure, I'll remove the pathways and incorporate a roadmap section as you suggested. I'll also introduce a challenges section. Could you please provide further details on what should be included in the challenges? Specifically, based on the content of the 'top 25 coding challenges' list, what elements should we retain, remove, or add?

**Comment by SkSadaf** [comment]
I have added the roadmap section, please review it and provide feedback.

**Comment by jfmartinz** [comment]
> Sure, I'll remove the pathways and incorporate a roadmap section as you suggested. I'll also introduce a challenges section. Could you please provide further details on what should be included in the challenges? Specifically, based on the content of the 'top 25 coding challenges' list, what elements should we retain, remove, or add?

The challenges section is something like [this](https://github.com/jfmartinz/ResourceHub/tree/main/Web%20Development#challenges) under web development category. But the difference is that it should be in Web3, I checked some of  the resources added, some are broken link or not working, not secured, 404 page, and some are not suited to be added in challenges.

**Comment by SkSadaf** [comment]
> > Sure, I'll remove the pathways and incorporate a roadmap section as you suggested. I'll also introduce a challenges section. Could you please provide further details on what should be included in the challenges? Specifically, based on the content of the 'top 25 coding challenges' list, what elements should we retain, remove, or add?
> 
> The challenges section is something like [this](https://github.com/jfmartinz/ResourceHub/tree/main/Web%20Development#challenges) under web development category. But the difference is that it should be in Web3, I checked some of the resources added, some are broken link or not working, not secured, 404 page, and some are not suited to be added in challenges.
> 
> You can take a look at the comment above, in which resources to be deleted or added

Okay, Thank you, I got an idea of what must be included. I will work on adding challenges.

**Comment by jfmartinz** [comment]
> > > Sure, I'll remove the pathways and incorporate a roadmap section as you suggested. I'll also introduce a challenges section. Could you please provide further details on what should be included in the challenges? Specifically, based on the content of the 'top 25 coding challenges' list, what elements should we retain, remove, or add?
> > 
> > 
> > The challenges section is something like [this](https://github.com/jfmartinz/ResourceHub/tree/main/Web%20Development#challenges) under web development category. But the difference is that it should be in Web3, I checked some of the resources added, some are broken link or not working, not secured, 404 page, and some are not suited to be added in challenges.
> > You can take a look at the comment above, in which resources to be deleted or added
> 
> Okay, Thank you, I got an idea of what must be included. I will work on adding challenges.

Thanks @SkSadaf. Ill wait for your changes.

**Comment by SkSadaf** [comment]
I have added the challenges. Please review and provide provide a feedback on this and the roadmap as well.

File: Web3/README.md [hunk 1]
```diff
@@ -378,3 +378,14 @@ Analytics.</td>
...
+|----------------------------|-----------------------------------------------------------------------------------------------------|
+| [Ethernaut](https://ethernaut.openzeppelin.com/)                  | Ethernaut is a Solidity-based smart contract hacking game. Each level presents a different smart contract vulnerability to exploit and solve. |
+| [Capture the Ether](https://capturetheether.com/challenges/)       | Capture the Ether offers a series of challenges designed to test and improve your skills in hacking Ethereum smart contracts. Solve challenges related to security vulnerabilities and earn Ether rewards. |
+| [Solidity Koans](https://soliditykoans.org/)                     | Solidity Koans is an interactive learning platform that presents Solidity code snippets with incomplete parts. Your task is to fill in the blanks to make the code functional and learn Solidity concepts along the way. |
+| [Blockchain CTF](https://ctftime.org/task/11566)   | Blockchain CTF (Capture The Flag) challenges offer a range of tasks related to blockchain technology, including Ethereum smart contract hacking, blockchain analysis, and cryptography. |
```
Comment (jfmartinz) [inline Web3/README.md:389]: Please foollow the format:

``` html
### Section name
<table width="100%">
 <tr>
   <th>Resource Name</th>
   <th>Description</th>
 </tr>
 <tr>
   <td><a href="URL HERE">RESOURCE NAME</a></td>
   <td>RESOURCE DESCRIPTION</td>
 </tr>
</table>
```

**Review by jfmartinz — CHANGES_REQUESTED** [review]

**Comment by jfmartinz** [comment]
Last one request anad I will merged it now, thans

**Comment by SkSadaf** [comment]
Sorry for the oversight, now I have made changes as per the format.

**Review by jfmartinz — APPROVED** [review]

**Pull request merged**